protocol/development-workflow/README.md

```diff
@@ -0,0 +1,125 @@
+Development Workflow
+====================
+
+Setting Up the Release Branch
+-----------------------------
+
+As a team, determine the scope of a release. The release can contain one
+Pivotal Tracker story, or it can combine multiple stories that cannot be
+deployed independently of one another.
+
+Create a release branch off of `master`.
+
+    git checkout master
+    git pull
+    git checkout -b <release-name>
+
+Push the release branch to GitHub.
+
+    git push -u origin <release-name>
+
+Developing for the Release
+--------------------------
+
+Create a development branch off of the release branch.
+
+    git checkout <release-name>
+    git pull
+    git checkout -b <development-branch-name>
+
+Commit changes.
+
+Rebase over the release branch.
+
+    git fetch origin
+    git rebase origin/<release-name>
+
+Push the development branch to GitHub.
+
+    git push -u origin <development-branch-name>
+
+Submit a pull request to merge into the release branch.
+
+Conduct Code Review.
+
+Merge the development branch into the release branch.
+
+Follow the above process as many times as necessary to complete all
+components of the release in development branches.
+
+Conducting QA
+-------------
+
+Get the most recent state of the release branch.
+
+    git checkout <release-name>
+    git pull
+
+Create a staging branch off of `master`.
+
+    git checkout master
+    git pull
+    git checkout -b stage-<release-name>
+
+Merge the release branch into the release's staging branch.
+
+    git merge <release-name>
+
+Deliver the staging branch to the staging deployment.
+
+    git push staging master
+
+Wait for the deployment to finish.
+
+Conduct QA through the staging app front-end.
+
+Rejecting
+---------
+
+Rollback the database migrations of this release on staging.
+
+Reset the staging remote.
+
+    git checkout master
+    git push staging master
+
+Delete the release's staging branch.
+
+    git branch -D stage-<release-name>
+
+Indicate in Pivotal Tracker that the release was rejected.
+
+Accepting and Deploying
```

Comment:
@KrishnaKulkarni are you suggesting that anything merged into the release branch gets automatically deployed to staging? I would like that very much. If that's the case, would the release branch always have the same name (staging?) to accommodate third party software that might expect the same name for all of the automation processes?